### PR TITLE
Feature/blacklisting

### PR DESCRIPTION
**Pull request by thomasam97**
based on #25
addresses #5 by doing so also #22 (the latter closed due to duplication)

### CODE REVIEW

**Comment by L4B0MB4** [comment]
I totally forgot in the last pullrequest. But i think we should extends the tests to test the new functionality. @thomasam97 @N3ol4N agree?

**Comment by thomasam97** [comment]
after a talk with @L4B0MB4 we setteled on deleting the testing branch. The tests would only test git log functionality.

File: src/modules/git/index.js [hunk 1]
```diff
@@ -19,19 +19,25 @@ export default function gLog(path, doneCB, queryParameter, fileExtension) {
...
+    const fileExtensionArray = fileExtension.split(',');
+    const prefixString = prefix.concat(fileExtensionArray.join(',*.'));
+    const extensions = prefixString.split(',');
+    cmdArgs = cmdArgs.concat(extensions);
+  } else {
+    const prefix = '';
+    const prefix2 = prefix.concat('--,', '.,', ':^*.')
+    const fileExtentionExclusionArray = fileExtentionExclusion.split(',');
+    const prefixString = prefix2.concat(fileExtentionExclusionArray.join(',:^*.'));
+    const exclusions = prefixString.split(',');
```
Comment (L4B0MB4) [inline src/modules/git/index.js]: This is unnecessary. You are changing types constantly. From string to array to string to array :) should be possible with only one or two arrays

File: src/views/start/index.jsx [hunk 2]
```diff
@@ -168,7 +181,8 @@ export default class Start extends Component {
     return (
       <div className="Start">
         <div>
-          Query Parameter :
+          Query parameter:
+          &nbsp;
```
Comment (L4B0MB4) [inline src/views/start/index.jsx]: In react you can write {" "} ... But one single space should be possible by just placing one behind the colons. Not 100% sure

File: src/views/start/index.jsx [hunk 3]
```diff
@@ -121,6 +122,18 @@ export default class Start extends Component {
...
+  onFileExtensionExclusionKeyDown = (e) => {
+    if (e.keyCode === 13) {
+      if (this.currentPath) {
+        gLog(this.currentPath, this.gLogDoneCB, this.queryValue, this.fileExtension, this.fileExtensionExclusion);
+      }
+    }
+  }
+
+  setFileExtensionExclusionValue = (e) => {
+    this.fileExtensionExclusion = e.target.value;
```
Comment (L4B0MB4) [inline src/views/start/index.jsx]: Reminder for me: is this logging a warning? Controlled uncontrolled input? Check that -> @L4B0MB4  :p

File: src/views/start/index.jsx (hunk 2 above)
Comment (thomasam97) [reply src/views/start/index.jsx]: This works great 👍

**Pull request merged**